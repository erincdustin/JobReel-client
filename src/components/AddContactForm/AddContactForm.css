.AddContactForm {
  border: 1px black solid;
  min-width: 300px;
  margin: 0 auto;
  padding: 5%;
  line-height: 150%;
  border-radius: 10px;
  text-align: center;
  background: #3b3a50;
  color: #eeebd9;
  grid-column-start: 6;
  grid-column-end: 9;
  box-shadow:  0.5em 0.5em 0.2em rgb(48, 39, 39);
}

#comments {
  width: 15em;
  border-radius: 10px;
}

@media screen and (max-width: 960px) {
  .AddContactForm {
    grid-column-start: 1;
    grid-column-end: 9;
  }
}
