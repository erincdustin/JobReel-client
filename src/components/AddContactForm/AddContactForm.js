import React from 'react';
import { Input, Label } from '../../components/Form/Form';
import Button from '../../components/Button/Button';
import JobReelContext from '../../context/JobReelContext';
import jobReelApiService from '../../services/jobreel-api-service';
import './AddContactForm.css';

class AddContactForm extends React.Component {

  static contextType = JobReelContext

  state = {
    error: null
  }

  handleSubmit = (e) => {
    e.preventDefault();
    e.persist();
    this.setState({ error: null });
    const contact_name = e.target.contact_name.value;
    const job_title = e.target.job_title.value;
    const company = e.target.company.value;
    const email = e.target.email.value;
    const linkedin = e.target.linkedin.value;
    const comments = e.target.comments.value;
    let connected;
    (e.target.connected.value === 'false') ? connected = false : connected = true;
    const userInput = { user_id: this.context.user.id, job_title, company, contact_name, email, linkedin, comments, connected };
    jobReelApiService.submitContact(userInput)
      .then(res => {
        e.target.contact_name.value = '';
        e.target.job_title.value = '';
        e.target.company.value = '';
        e.target.email.value = '';
        e.target.linkedin.value = '';
        e.target.comments.value = '';
        e.target.connected.value = false;
        this.context.setContacts([...this.context.contacts, res]);
        this.context.setManualContactAdd(false)
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }

  render() {
    const { error } = this.state;
    return (
      <div className="AddContactForm">
        <div role='alert'>
          {error && <p>{error}</p>}
        </div>
        <form onSubmit={this.handleSubmit}>
          <div>
            <Label htmlFor='contact-name-input'>
              Contact
            </Label>
            <br/>
            <Input
              id='contact-name-input'
              name='contact_name'
            />
          </div>
          <div>
            <Label htmlFor='job-title-input'>
              Job Title
            </Label>
            <br/>
            <Input
              id='job-title-input'
              name='job_title'
            />
          </div>
          <div>
            <Label htmlFor='company-input'>
              Company
            </Label>
            <br/>
            <Input
              id='company-input'
              name='company'
            />
          </div>
          <div>
            <Label htmlFor='email-input'>
              Email
            </Label>
            <br/>
            <Input
              id='email-input'
              name='email'
            />
            <p>Need to verify email? Click <a target="_blank" rel="noopener noreferrer" href="https://hunter.io/email-verifier">here</a></p>
          </div>
          <div>
            <Label htmlFor='linkedin-input'>
              LinkedIn
            </Label>
            <br/>
            <Input
              id='linkedin-input'
              name='linkedin'
            />
          </div>
          <div>
            <Label htmlFor='comments-input'>
              Comments
            </Label>
            <br/>
            <Input
              id='comments-input'
              name='comments'
            />
          </div>
          <div>
          <Label htmlFor='connected'>Connection Status</Label>
          <select
              id='connected-Input'
              name='connected'
            >
              <option value="false">Not Connected</option>
              <option value="true">Connected</option>
            </select>
        </div>
          <Button onClick={() => this.context.setManualContactAdd(false)} type="button">Back</Button>
          <Button type="submit">Submit</Button>
        </form>
      </div>
    )
  }
}

export default AddContactForm;