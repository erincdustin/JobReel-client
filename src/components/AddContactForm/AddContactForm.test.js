import React from 'react'
import ReactDOM from 'react-dom'
import AddContactForm from './AddContactForm'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('AddContactForm smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AddContactForm />, div);
    ReactDOM.unmountComponentAtNode(div);
  })
})

describe('AddContactForm snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(<AddContactForm />)
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})