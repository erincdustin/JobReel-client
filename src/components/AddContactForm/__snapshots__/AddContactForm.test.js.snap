// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddContactForm snapshot test renders the UI as expected 1`] = `
<div
  className="AddContactForm"
>
  <div
    role="alert"
  />
  <form
    onSubmit={[Function]}
  >
    <div>
      <Label
        htmlFor="contact-name-input"
      >
        Contact
      </Label>
      <br />
      <ForwardRef
        id="contact-name-input"
        name="contact_name"
      />
    </div>
    <div>
      <Label
        htmlFor="job-title-input"
      >
        Job Title
      </Label>
      <br />
      <ForwardRef
        id="job-title-input"
        name="job_title"
      />
    </div>
    <div>
      <Label
        htmlFor="company-input"
      >
        Company
      </Label>
      <br />
      <ForwardRef
        id="company-input"
        name="company"
      />
    </div>
    <div>
      <Label
        htmlFor="email-input"
      >
        Email
      </Label>
      <br />
      <ForwardRef
        id="email-input"
        name="email"
      />
      <p>
        Need to verify email? Click 
        <a
          href="https://hunter.io/email-verifier"
          rel="noopener noreferrer"
          target="_blank"
        >
          here
        </a>
      </p>
    </div>
    <div>
      <Label
        htmlFor="linkedin-input"
      >
        LinkedIn
      </Label>
      <br />
      <ForwardRef
        id="linkedin-input"
        name="linkedin"
      />
    </div>
    <div>
      <Label
        htmlFor="comments-input"
      >
        Comments
      </Label>
      <br />
      <ForwardRef
        id="comments-input"
        name="comments"
      />
    </div>
    <div>
      <Label
        htmlFor="connected"
      >
        Connection Status
      </Label>
      <select
        id="connected-Input"
        name="connected"
      >
        <option
          value="false"
        >
          Not Connected
        </option>
        <option
          value="true"
        >
          Connected
        </option>
      </select>
    </div>
    <ForwardRef
      onClick={[Function]}
      type="button"
    >
      Back
    </ForwardRef>
    <ForwardRef
      type="submit"
    >
      Submit
    </ForwardRef>
  </form>
</div>
`;
